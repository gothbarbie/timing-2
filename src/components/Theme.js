import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
