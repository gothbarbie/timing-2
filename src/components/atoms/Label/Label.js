import styled from 'styled-components'

const Label = styled.label`
  font-size: 1.4rem;
  font-weight: 500;
  color: ${({ theme }) => theme.colors.dark};
`

export default Label
