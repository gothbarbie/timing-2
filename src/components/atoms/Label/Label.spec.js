import React from 'react'
import { shallow } from 'enzyme'

import { theme } from '../../Theme'
import Label from './Label'

describe('generic/atoms/Label', () => {
  it('renders', () => {
    const component = shallow(<Label theme={theme}>Text</Label>)

    expect(component).toMatchSnapshot()
  })
})
