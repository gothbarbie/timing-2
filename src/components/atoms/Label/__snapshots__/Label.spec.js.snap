// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/atoms/Label renders 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-size: 1.4rem;
  font-weight: 500;
  color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.label",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "label",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "colors": Object {
        "background": "#EEEEEE",
        "black": "#444444",
        "dark": "#888888",
        "gray": "#BBBBBB",
        "lightGray": "#F6F6F6",
        "primary": "#6A9AE2",
        "primaryLight": "#E7F0FF",
        "white": "#FCFCFC",
      },
    }
  }
>
  Text
</StyledComponent>
`;
